thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -203,7 +253,22 @@ class CIndexReadSlaveBase : public CSlaveActivity
                             }
                         }
                         else
-                            remoteCandidates.push_back(copy);
+                        {
+                            if ((0 == avgPartSize) && inChildQuery() && !getOptBool(THOROPT_STREAM_REMOTE_SMALL_INDEX)) // if file size from meta was unavailable
```

Comment:
I think this might mean some parts are streamed remote and some are executed locally? It doesn't feel like the decision is in the correct place.

Reply:
agree, will move so calculated once.
Although the existing code still does allow for some to be streamed and some not, depending on whether the parts have different formats and are/are not translatable. e.g. could be a super with different formats, where some are are translatable and therefore streamable, and others are not.